src/netlink_namespace.c

```diff
@@ -719,36 +730,35 @@ static int ir_init (void)
   DEBUG("netlink_namespace plugin: ir_init: Switching to namespace on file '%s'.", ns_name);
   if ((targetns = open(ns_name, O_RDONLY)) < 0) {
     ERROR("Unable to open namespace file '%s'", ns_name);
-    close_fds(&selfns, &targetns);
+    ir_init_error(&selfns, &targetns);
     return -1;
   }
 
   if (setns(targetns, CLONE_NEWNET) != 0) {
     ERROR("netlink_namespace plugin: ir_init: Unable to switch namespace to '%s'", ns_name);
-    close_fds(&selfns, &targetns);
-    return -1;
+    ir_init_error(&selfns, &targetns);
+    abort();
```

Comment:
Wrong place to abort, at this point, it's perfectly fine to return -1. We haven't mutated any state we cannot revert.